app/controllers/admin/users_controller.rb

```diff
@@ -35,11 +35,15 @@ def index
         search: params[:search],
         role: params[:role],
         roles: params[:roles],
+        joining_start: params[:joining_start],
+        joining_end: params[:joining_end],
+        date_format: params[:date_format],
```

Comment:
Reading note:  I'm beginning to wonder if we want to use `**params.slice(:search, :role, :roles, :joining_start, :joining_end, :date_format).symbolize_keys`; the reason is that any missing keys won't be sent as keyword arguments, thus not clobbering the default argument values of `Admin::UsersQuery.call`

See the below example on `splice`; and the prefixed `**` converts the hash into keyword arguments.

```ruby
[6] pry(main)> { a: 1, b: 2, c: 3 }.slice(:a, :d)
=> {:a=>1}
```

Reply:
that does seem nicer 👍

Reply:
Alas I'm still having issues:

```
Unpermitted parameters: :utf8, :roles, :locale. Context: { controller: Admin::UsersController, action: index, request: #<ActionDispatch::Request:0x00007fc4abac3ab0>, params: {"utf8"=>"✓", "roles"=>["Admin"], "joining_start"=>"", "joining_end"=>"", "date_format"=>"DD/MM/YYYY", "locale"=>nil, "controller"=>"admin/users", "action"=>"index"} }
```

I honestly don't know why it's saying `roles` is unpermitted (does the same if I try to filter by organization etc).

This is what I have:

```rb
    ATTRIBUTES_FOR_USER_INDEX = %i[search role roles joining_start joining_end date_format organizations].freeze

     kwargs = params.permit(ATTRIBUTES_FOR_USER_INDEX).to_h.slice(ATTRIBUTES_FOR_USER_INDEX)
     @users = Admin::UsersQuery.call(relation: User.registered, **kwargs).page(params[:page]).per(50)
```

Reply:
I've just reverted this back for now as it was still causing headaches. If/when we continue work with the filters I'll look to refactor it 👍